src/Tests/Libs/GirTest-0.1.Tests/UnhandledExceptionTest.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace GirTest.Tests;
+
+[TestClass, TestCategory("BindingTest")]
+public class UnhandledExceptionTest : Test
+{
+    [TestMethod]
+    public void SupportsSignalHandlers()
+    {
+        var tester = ReturningSignalTester.New();
+        tester.OnReturnBool += (_, _) =>
+        {
+            throw new NotImplementedException();
+        };
+
+        bool exceptionCaught = false;
+        GLib.UnhandledException.SetHandler(e =>
```

Comment:
Just a note for future changes - if tests can run in parallel we need to be careful about this when more test cases are added for other callback types
Maybe we should leave a comment in the code to mention this?

Reply:
I added a comment.